src/Features/CSharp/Portable/CodeRefactorings/InlineMethod/CSharpInlineMethodRefactoringProvider.cs

```diff
@@ -0,0 +1,157 @@
+            {
+                return identifierNameSyntax.Identifier.ValueText;
+            }
+
+            return string.Empty;
+        }
+
+        protected override SyntaxNode GenerateArrayInitializerExpression(ImmutableArray<SyntaxNode> arguments)
+            => SyntaxFactory.InitializerExpression(SyntaxKind.ArrayInitializerExpression, SyntaxFactory.SeparatedList(arguments));
+
+        protected override SyntaxNode GenerateLocalDeclarationStatementWithRightHandExpression(
+            string identifierTokenName,
+            ITypeSymbol type,
+            SyntaxNode expression)
+            => SyntaxFactory.LocalDeclarationStatement(
+                SyntaxFactory.VariableDeclaration(
+                    type.GenerateTypeSyntax(),
+                    SyntaxFactory.SingletonSeparatedList(
+                        SyntaxFactory.VariableDeclarator(
+                            SyntaxFactory.Identifier(identifierTokenName),
+                            argumentList: null,
+                            initializer: SyntaxFactory.EqualsValueClause((ExpressionSyntax)expression)))));
+
+        protected override SyntaxNode GenerateLocalDeclarationStatement(string identifierTokenName, ITypeSymbol type)
+            => SyntaxFactory.LocalDeclarationStatement(
+                SyntaxFactory.VariableDeclaration(
+                    type.GenerateTypeSyntax(),
+                    SyntaxFactory.SingletonSeparatedList(SyntaxFactory.VariableDeclarator(identifierTokenName))));
+
+        protected override SyntaxNode GenerateIdentifierNameSyntaxNode(string name)
+            => SyntaxFactory.IdentifierName(name);
+
+        protected override SyntaxNode? GetInlineStatement(SyntaxNode calleeMethodDeclarationSyntaxNode)
+        {
+            SyntaxNode? inlineSyntaxNode = null;
+            if (calleeMethodDeclarationSyntaxNode is MethodDeclarationSyntax declarationSyntax)
+            {
+                var blockSyntaxNode = declarationSyntax.Body;
+                // 1. If it is a ordinary method with block
+                if (blockSyntaxNode != null)
+                {
+                    var blockStatements = blockSyntaxNode.Statements;
+                    if (blockStatements.Count == 1)
+                    {
+                        inlineSyntaxNode = GetExpressionFromStatementSyntaxNode(blockStatements[0]);
+                    }
+                }
+                else
+                {
+                    // 2. If it is using Arrow Expression
+                    var arrowExpressionNodes = declarationSyntax
+                        .DescendantNodes().Where(node => node.IsKind(SyntaxKind.ArrowExpressionClause)).ToImmutableArray();
+                    if (arrowExpressionNodes.Length == 1)
+                    {
+                        inlineSyntaxNode = ((ArrowExpressionClauseSyntax)arrowExpressionNodes[0]).Expression;
+                    }
+                }
+            }
+
+            return inlineSyntaxNode;
```

Comment:
seems bad if we coudl hit this.

Reply:
Emm, it shouldn't hit here because the the check has been done before. I'll throw Unreachable Exception here